include/CXXGraph/Graph/Algorithm/isBipartiteGraph.hpp

```diff
@@ -0,0 +1,91 @@
+/***********************************************************/
+/***      ______  ____  ______                 _         ***/
+/***     / ___\ \/ /\ \/ / ___|_ __ __ _ _ __ | |__      ***/
+/***    | |    \  /  \  / |  _| '__/ _` | '_ \| '_ \     ***/
+/***    | |___ /  \  /  \ |_| | | | (_| | |_) | | | |    ***/
+/***     \____/_/\_\/_/\_\____|_|  \__,_| .__/|_| |_|    ***/
+/***                                    |_|             ***/
+/***********************************************************/
+/***     Header-Only C++ Library for Graph               ***/
+/***   Representation and Algorithms                     ***/
+/***********************************************************/
+/***     Author: ZigRazor ***/
+/***    E-Mail: zigrazor@gmail.com                       ***/
+/***********************************************************/
+/***    Collaboration: -----------                       ***/
+/***********************************************************/
+/***    License: AGPL v3.0 ***/
+/***********************************************************/
+
+#ifndef __CXXGRAPH_CHECK_ODD_NODES_CYCLE_H__
+#define __CXXGRAPH_CHECK_ODD_NODES_CYCLE_H__
+
+#pragma once
+
+#include "CXXGraph/Graph/Graph_decl.h"
+
+namespace CXXGraph {
+
+template <typename T>
+class Graph {
+ public:
+  TopoSortResult<T> checkOddNodesCycleDir() const;
+
+ private:
+  void checkOddNodesCycleUtil(std::vector<int>& visited,
+                              std::vector<int>& chainNo, int root,
+                              int cycleLength, bool& isBiPartite) const;
+};
```

Comment:
You can't re-declare the class here. Declare the methods in the Graph_decl.hpp file, possibly adding method descriptions like you see there, and in this file you only define them